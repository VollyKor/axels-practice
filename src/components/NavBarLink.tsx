import { LinkContainer } from 'react-router-bootstrap';
import { Nav } from 'react-bootstrap';

interface Props {
    children: React.ReactNode;
    to: string;
}

const NavBarLink = ({ children, to }: Props) => (
    <LinkContainer to={to} exact>
        <Nav.Link>{children}</Nav.Link>
    </LinkContainer>
);

export default NavBarLink;
