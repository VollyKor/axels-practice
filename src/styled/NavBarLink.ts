import styled from 'styled-components';

import NavBarLink from 'components/NavBarLink';

export const NavLink = styled(NavBarLink)`
    color: var(--logo-color);
`;
