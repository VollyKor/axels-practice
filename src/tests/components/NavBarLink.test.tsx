import { shallow } from 'enzyme';

import NavBarLink from 'components/NavBarLink';

it('renders snapshot', () => {
    const wrapper = shallow(<NavBarLink children to={'/'} />);
    expect(wrapper).toMatchSnapshot();
});
