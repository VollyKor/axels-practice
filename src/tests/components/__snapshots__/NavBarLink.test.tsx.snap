// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<withRouter(LinkContainer)
  exact={true}
  to="/"
>
  <NavLink
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    disabled={false}
  >
    <Component />
  </NavLink>
</withRouter(LinkContainer)>
`;
